### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by jianyunt**
Raise error when a package does not exist, e.g., install-NanoServerPackage foobar
TFS#8912396 - improve error message for versions  
TFS# 8912355 - remove unused -force
TFS# 8912448- remove unused -norestart
Fixed test failures

### CODE REVIEW

**Comment by jianyunt** [comment]
@KrishnaV-MSFT @bmanikm, please perform code review. thanks. #Resolved

**Comment by bmanikm** [comment]
> #>

Please remove this comment too. #Closed

---

Refers to: NanoServerPackage/NanoServerPackage.psm1:415 in 772bd77. [](commit_id = 772bd77c45696ab0ca09783a7cef573b227a743f, deletion_comment = False)

File: NanoServerPackage/NanoServerPackage.psm1 [hunk 1]
```diff
@@ -453,14 +444,20 @@ function Install-NanoServerPackage
...
         if ($packagesToBeInstalled.Count -eq 0)
-        {
-            return
+        { 
+            ThrowError -CallerPSCmdlet $PSCmdlet `
+                -ExceptionName System.Exception `
+                -ExceptionMessage ("Package '{0}' not found" -f $Name) `
+                -ExceptionObject $packageName `
+                -ErrorId PackageNotFound `
+                -ErrorCategory InvalidData
```
Comment (bmanikm) [inline NanoServerPackage/NanoServerPackage.psm1]: > InvalidData [](start = 31, length = 11)

InvalidOperation? #Closed

File: NanoServerPackage/NanoServerPackage.psm1 [hunk 2]
```diff
@@ -453,14 +444,20 @@ function Install-NanoServerPackage
...
         $packagesToBeInstalled += (Find -Name $Name -MinimumVersion $MinimumVersion -MaximumVersion $MaximumVersion -RequiredVersion $RequiredVersion `
             -Culture $Culture -ErrorAction Stop) # -Repository $Repository
-
+      
         if ($packagesToBeInstalled.Count -eq 0)
-        {
-            return
+        { 
+            ThrowError -CallerPSCmdlet $PSCmdlet `
+                -ExceptionName System.Exception `
```
Comment (bmanikm) [inline NanoServerPackage/NanoServerPackage.psm1]: > Exception [](start = 38, length = 9)

System.InvalidOperationException ? #Closed

File: NanoServerPackage/NanoServerPackage.psm1 [hunk 3]
```diff
@@ -654,7 +652,6 @@ function Install-NanoServerPackage
                                                             -availablePackages $availablePackages `
                                                             -successfullyInstalled ([ref]$success) `
                                                             -Force:$Force `
-                                                            -NoRestart:$NoRestart `
                                                             -PackagesToBeInstalled $packagesToBeInstalled
 #-source $source `
```
Comment (bmanikm) [inline NanoServerPackage/NanoServerPackage.psm1]: Could you please remove the commented code in this file? #Closed

**Comment by bmanikm** [comment]
:shipit:

File: NanoServerPackage/NanoServerPackage.psm1 (hunk 1 above)
Comment (jianyunt) [reply NanoServerPackage/NanoServerPackage.psm1]: fixed

---

In reply to: [81068435](https://github.com/OneGet/NanoServerPackage/pull/5#discussion_r81068435) [](ancestors = 81068435)

File: NanoServerPackage/NanoServerPackage.psm1 (hunk 2 above)
Comment (jianyunt) [reply NanoServerPackage/NanoServerPackage.psm1]: fixed

---

In reply to: [81068506](https://github.com/OneGet/NanoServerPackage/pull/5#discussion_r81068506) [](ancestors = 81068506)

**Pull request merged**